gnovm/pkg/gnolang/go2gno.go

```diff
@@ -520,6 +535,9 @@ type gnoImporter struct {
 	getter MemPackageGetter
 	cache  map[string]gnoImporterResult
 	cfg    *types.Config
+
+	// if true, allows re-definitions
+	testing bool
```

Comment:
I am not a huge fan of adding testing code in objects used outside the testing suite, but I realize this would require a much larger rework to implement elegantly

Please resolve this convo

Reply:
Note that it's not just about our own testing suite, but also about `gno test`, which is a key feature of the `gno` tool

I've changed the struct field to be called `allowRedefitions` instead, so we can decouple the logic for redefinitions and more general "what changes in testing" (though it should be an ever-shrinking difference; but should make what's happening a bit more explicit).